src/robusta/core/sinks/sink_base.py

```diff
@@ -1,12 +1,29 @@
-from typing import Any
+import threading
+from abc import abstractmethod, ABC
+from collections import defaultdict
+from typing import Any, List, Dict, Tuple, DefaultDict
 
 from robusta.core.model.k8s_operation_type import K8sOperationType
 from robusta.core.reporting.base import Finding
 from robusta.core.sinks.sink_base_params import ActivityInterval, ActivityParams, SinkBaseParams
 from robusta.core.sinks.timing import TimeSlice, TimeSliceAlways
 
 
-class SinkBase:
+class SinkBase(ABC):
+    grouping_enabled: bool
+    grouping_summary_mode: bool
+
+    # The tuples in the types below holds all the attributes we are aggregating on.
+    finding_group_start_ts: Dict[Tuple, float]  # timestamps for message groups
+    finding_group_n_received: DefaultDict[Tuple, int]  # number of messages ignored for each group
+    finding_group_heads: Dict[Tuple, str]  # a mapping from a set of parameters to the head of a thread
+
+    # Summary groups
+    finding_summary_header: List[str]  # descriptive header for the summary table
+    finding_summary_counts: DefaultDict[Tuple, DefaultDict[Tuple, List[int]]]  # rows of the summary table
+
+    finding_group_lock: threading.Lock = threading.RLock()
```

Comment:
should this be a class member?

Reply:
Yeah, no need to lock *all* Slack sinks. Changed to an instance member.